### ISSUE TITLE
Icon Tooltip: AVT 1 violation

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3094]
## Environment

> Operating system

Mac OS X

> Browser

All

> Automated testing tool and ruleset

AAT v4.2.1

## Detailed description

Getting error from AAT test suite: `"wcag20.tech.g1204.inputVisibleLabel": "The input component is missing an associated visible label."`

> What version of the Carbon Design System are you using?

v10.3.1

> What did you expect to happen?

Not to get any AAT violations

> What happened instead?

Got a violation

## Steps to reproduce the issue

1. run the latest test suite of AAT on icon tooltip

## Additional information

This can be resolved by using a span within the button and making the text visually hidden in the default state and making it visible on focus/hover.  See Watson Health's demo for an example: https://pages.github.ibm.com/Watson-Health/components/detail/button--icon-only.html
